cabal-install/tests/UnitTests/Distribution/Client/Init/FileCreators.hs

```diff
@@ -0,0 +1,152 @@
+module UnitTests.Distribution.Client.Init.FileCreators (
+  tests
+  ) where
+
+import Distribution.Client.Init.FileCreators
+  ( generateCabalFile )
+
+import Test.Tasty
+import Test.Tasty.Golden (goldenVsString)
+
+import System.FilePath
+  ( (</>) )
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as BS8
+import qualified Data.Set as Set
+
+import Distribution.Client.Init.Types
+  ( InitFlags(..), PackageType(..) )
+import Distribution.Simple.Setup
+  ( Flag(..) )
+
+import Distribution.CabalSpecVersion
+  ( CabalSpecVersion(CabalSpecV2_4) )
+import Distribution.Types.Dependency
+  ( Dependency, mkDependency )
+import Distribution.Types.LibraryName
+  ( LibraryName(LMainLibName) )
+import Distribution.Types.PackageName
+  ( mkPackageName )
+import Distribution.Types.VersionRange
+  ( majorBoundVersion )
+import Distribution.Types.Version
+  ( mkVersion )
+import qualified Distribution.ModuleName as ModuleName
+  ( fromString )
+import qualified Distribution.SPDX as SPDX
+import Language.Haskell.Extension ( Language(..) )
+
+tests :: [TestTree]
+tests = [ testGroup "cabal init goldens"
+          [ checkCabalFileGolden exeFlags "exe-only-golden.cabal"
+          , checkCabalFileGolden libExeAndTestFlags "lib-exe-and-test-golden.cabal"
+          ]
+        ]
+
+checkCabalFileGolden :: InitFlags -> FilePath -> TestTree
+checkCabalFileGolden flags goldenFileName =
+  goldenVsString goldenFileName goldenFilePath generatedCabalFile
+  where
+    goldenFilePath :: FilePath
+    goldenFilePath = "tests" </> "fixtures" </> "init" </> goldenFileName
+
+    generatedCabalFile :: IO BS.ByteString
+    generatedCabalFile = pure . BS8.pack $ generateCabalFile goldenFileName flags
+
+-- ==================================================
+-- Base flags to set common InitFlags values.
+
+baseFlags :: InitFlags
+baseFlags = InitFlags {
```

Comment:
Ah, sorry. I'd still like this to use `defaultInitFlags`, but I'll could make that change myself.

I want to see "how much" non-defaults there are.

Reply:
Ahhh gotcha, switched to use `defaultInitFlags` as the base (it only sets `initVerbosity` which I've removed from the list here).